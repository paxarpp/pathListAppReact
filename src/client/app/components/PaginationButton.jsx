import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class PaginationButton extends Component {
  hendler = page => () => {
    const { handlerPagination } = this.props;
    handlerPagination(page);
  };

  render() {
    const { length, page, stringOnPage } = this.props;
    const pages =
      length % stringOnPage === 0
        ? length / stringOnPage
        : Math.ceil(length / stringOnPage);

    return (
      <div className="paginationBlock">
        {page <= 1 ? null : <button onClick={this.hendler(1)}>1</button>}
        {page < 5 ? null : <span>...</span>}
        {page - 10 <= 1 ? null : (
          <button onClick={this.hendler(page - 10)}>{page - 10}</button>
        )}
        {page - 2 <= 1 ? null : (
          <button onClick={this.hendler(page - 2)}>{page - 2}</button>
        )}
        {page - 1 <= 1 ? null : (
          <button onClick={this.hendler(page - 1)}>{page - 1}</button>
        )}
        {pages <= 1 ? null : (
          <button onClick={this.hendler(page)} className="selectPagination">
            {page}
          </button>
        )}
        {page + 1 >= pages ? null : (
          <button onClick={this.hendler(page + 1)}>{page + 1}</button>
        )}
        {page + 2 >= pages ? null : (
          <button onClick={this.hendler(page + 2)}>{page + 2}</button>
        )}
        {page + 10 >= pages ? null : (
          <button onClick={this.hendler(page + 10)}>{page + 10}</button>
        )}
        {page > pages - 4 ? null : <span>...</span>}
        {page + 1 > pages ? null : (
          <button onClick={this.hendler(pages)}>{pages}</button>
        )}
      </div>
    );
  }
}
