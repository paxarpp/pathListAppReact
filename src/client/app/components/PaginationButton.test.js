import React from 'react';
import renderer from 'react-test-renderer';
import PaginationButton from './PaginationButton';

test('PaginationButton empty', () => {
  const length = 0;
  const page = 0;
  const stringOnPage = 20;
  const component = renderer.create(
    <PaginationButton length={length} page={page} stringOnPage={stringOnPage} handlerPagination={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('PaginationButton 100etc', () => {
  const length = 100;
  const page = 1;
  const stringOnPage = 20;
  const component = renderer.create(
    <PaginationButton length={length} page={page} stringOnPage={stringOnPage} handlerPagination={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('PaginationButton 500etc', () => {
  const length = 500;
  const page = 10;
  const stringOnPage = 20;
  const component = renderer.create(
    <PaginationButton length={length} page={page} stringOnPage={stringOnPage} handlerPagination={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('PaginationButton 400etc', () => {
  const length = 400;
  const page = 20;
  const stringOnPage = 20;
  const component = renderer.create(
    <PaginationButton length={length} page={page} stringOnPage={stringOnPage} handlerPagination={() => {}} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
