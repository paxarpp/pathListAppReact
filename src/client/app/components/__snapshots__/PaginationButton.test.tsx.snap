// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationButton 100etc 1`] = `
<div
  className="sc-bxivhb kqodyu"
>
  <button
    className="sc-ifAKCX bLOlJY sc-bwzfXH bFVykN"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    3
  </button>
  <span>
    ...
  </span>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    5
  </button>
</div>
`;

exports[`PaginationButton 400etc 1`] = `
<div
  className="sc-bxivhb kqodyu"
>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    1
  </button>
  <span>
    ...
  </span>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    19
  </button>
  <button
    className="sc-ifAKCX bLOlJY sc-bwzfXH bFVykN"
    onClick={[Function]}
  >
    20
  </button>
</div>
`;

exports[`PaginationButton 500etc 1`] = `
<div
  className="sc-bxivhb kqodyu"
>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    1
  </button>
  <span>
    ...
  </span>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    9
  </button>
  <button
    className="sc-ifAKCX bLOlJY sc-bwzfXH bFVykN"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    11
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    12
  </button>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    20
  </button>
  <span>
    ...
  </span>
  <button
    className="sc-bwzfXH ixPAYQ"
    onClick={[Function]}
  >
    25
  </button>
</div>
`;

exports[`PaginationButton empty 1`] = `
<div
  className="sc-bxivhb kqodyu"
/>
`;
